src/main/java/edu/cornell/kfs/pdp/batch/service/impl/PayeeACHAccountExtractServiceImpl.java

```diff
@@ -0,0 +1,702 @@
+
+    private BatchInputFileService batchInputFileService;
+    private List<BatchInputFileType> batchInputFileTypes;
+    private ParameterService parameterService;
+    private MailService mailService;
+    private DocumentService documentService;
+    private DataDictionaryService dataDictionaryService;
+    private PersonService personService;
+    private SequenceAccessorService sequenceAccessorService;
+    private AchService achService;
+    private AchBankService achBankService;
+
+    // Portions of this method are based on code and logic from CustomerLoadServiceImpl.
+    @Transactional
+    @Override
+    public boolean processACHBatchDetails() {
+        LOG.info("Beginning processing of ACH input files");
+        
+        int numSuccess = 0;
+        int numPartial = 0;
+        int numFail = 0;
+        
+        // create a list of the files to process.
+        Map<String,BatchInputFileType> fileNamesToLoad = getListOfFilesToProcess();
+        LOG.info("Found " + fileNamesToLoad.size() + " file(s) to process.");
+        
+        // process each file in turn
+        List<String> processedFiles = new ArrayList<String>();
+        for (String inputFileName : fileNamesToLoad.keySet()) {
+            
+            LOG.info("Beginning processing of filename: " + inputFileName);
+            processedFiles.add(inputFileName);
+            
+            try {
+                if (loadACHBatchDetailFile(inputFileName, fileNamesToLoad.get(inputFileName))) {
+                    LOG.info("Successfully loaded ACH input file");
+                    numSuccess++;
+                } else {
+                    LOG.warn("ACH input file contained one or more rows that could not be processed");
+                    numPartial++;
+                }
+            } catch (Exception e) {
+                LOG.error("Failed to load ACH input file", e);
+                numFail++;
+            }
+        }
+        
+        removeDoneFiles(processedFiles);
+        
+        LOG.info("==== Summary of Payee ACH Account Extract ====");
+        LOG.info("Files loaded successfully: " + Integer.toString(numSuccess));
+        LOG.info("Files loaded with one or more failed rows: " + Integer.toString(numPartial));
+        LOG.info("Files with errors: " + Integer.toString(numFail));
+        LOG.info("==== End Summary ====");
+        
+        // For now, return true even if files or rows did not load successfully. Functionals will address the failed rows/files accordingly.
+        return true;
+    }
+
+    /**
+     * Create a collection of the files to process with the mapped value of the BatchInputFileType
+     */
+    // This is a slightly-tweaked copy of a method from CustomerLoadServiceImpl.
+    protected Map<String,BatchInputFileType> getListOfFilesToProcess() {
+        Map<String,BatchInputFileType> inputFileTypeMap = new LinkedHashMap<String, BatchInputFileType>();

```

Comment:
Not sure if this is an issue, but the calling code is iterating over this Map's keys and also calling its get() method within the loop (and LinkedHashMap is modified structurally when get() is called). There are numerous other areas in KFS where this iteration is used, including the CustomerLoadServiceImpl processing that I had originally based portions of this service from, so I'm fine with it going out as-is.

If problems do arise from potential structural modification while iterating, we can address that in a separate user story so that we can update all of the affected services at once.